frontend/src/employee-frontend/components/unit/TabCalendar.tsx

```diff
@@ -406,3 +405,61 @@ const Caretakers = React.memo(function Caretakers({
     })
     .getOrElse(null)
 })
+
+type ActiveDateRangeSelectorProps = {
+  selectedDate: LocalDate
+  setSelectedDate: React.Dispatch<React.SetStateAction<LocalDate>>
+  mode: CalendarMode
+  weekRange: FiniteDateRange
+}
+
+const ActiveDateRangeSelector = React.memo(function ActiveDateRangeSelector({
+  selectedDate,
+  setSelectedDate,
+  mode,
+  weekRange
+}: ActiveDateRangeSelectorProps) {
+  const { i18n } = useTranslation()
+
+  const startDate =
+    mode === 'week' ? weekRange.start : selectedDate.startOfMonth()
+  const endDate = mode === 'week' ? weekRange.end : startDate.lastDayOfMonth()
+
+  const addUnitOfTime = useCallback(
+    (date: LocalDate) =>
+      mode === 'week' ? date.addDays(7) : date.addMonths(1),
+    [mode]
+  )
+  const subUnitOfTime = useCallback(
+    (date: LocalDate) =>
+      mode === 'week' ? date.subDays(7) : date.subMonths(1),
+    [mode]
+  )
+
+  return (
+    <>
+      <ColoredH3 noMargin>
+        {startDate.format('dd.MM.')}–{endDate.format('dd.MM.yyyy')}
```

Comment:
Would this be even better if it was just eg. "September 2022" in month mode instead of the date range?

Reply:
From a readability point of view, yes. But I tried this and it makes it awful to navigate more than one month: The month names are so different in length that the arrows jump _a lot_ making it impossible to click the same arrow twice and easy to go e.g. forward 1 and back again by mistake because the position of the arrows changed underneath the mouse cursor.

Thought about moving the arrows before the month name, but then it is more unclear what the arrows refer to. I think leaving the dates is the better option after all.

Reply:
Oh yeah, I had already forgotten about that. Some dates are longer than others, so they kind of have the same problem 😅. Could a `MM.yyyy` or `M.yyyy` formatted date work?